src/runtime/typemanager.cs

```diff
@@ -761,14 +878,43 @@ internal static void InitializeSlots(IntPtr type, Type impl)
         /// </summary>
         /// <param name="type">Type being initialized.</param>
         /// <param name="slot">Function pointer.</param>
-        /// <param name="name">Name of the method.</param>
-        static void InitializeSlot(IntPtr type, IntPtr slot, string name)
+        /// <param name="canOverride">Can override the slot when it existed</param>
+        static void InitializeSlot(IntPtr type, IntPtr slot, string name, bool canOverride = true)
```

Comment:
Can you explain why is it OK sometimes to not overwrite a slot?

I see this is used for `tp_traverse` and `tp_clear`. What are consequences of not overwriting this slot? Would not this cause our cleanup code to simply fail to be run?

Reply:
I think it relates to other parts of #958 where some classes define their own `tp_traverse` and `tp_clear`. This could be removed for now